code/game/objects/structures/industrial_lift.dm

```diff
@@ -0,0 +1,277 @@
+///Collect and command
+/datum/lift_master
+	var/list/lift_platforms
+
+/datum/lift_master/Destroy()
+	for(var/l in lift_platforms)
+		var/obj/structure/industrial_lift/lift_platform = l
+		lift_platform.LMaster = null
+	lift_platforms = null
+	return ..()
+
+/datum/lift_master/New(obj/structure/industrial_lift/lift_platform)
+	Rebuild_lift_plaform(lift_platform)
+
+///Collect all bordered platforms
+/datum/lift_master/proc/Rebuild_lift_plaform(obj/structure/industrial_lift/base_lift_platform)
+	LAZYOR(lift_platforms, base_lift_platform)
```

Comment:
Add and remove things from this list through procs, and register and unregister on their destruction, so if they get qdeleted early we don't keep a reference.
Examples:
```cpp
LAZYOR(lift_platforms, base_lift_platform)
```
becomes
```
add_lift_platforms(base_lift_platform)

/datum/lift_master/proc/add_lift_platforms(obj/structure/industrial_lift/new_lift_platform)
	if(new_lift_platform in lift_platforms)
		return
	LAZYADD(lift_platforms, new_lift_platform)
	RegisterSignal(new_lift_platform, COMSIG_PARENT_QDELETING, .proc/remove_lift_platforms)
```